extensions/amp-story/1.0/amp-story-interactive-poll.js

```diff
@@ -0,0 +1,134 @@
+ *
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildPollTemplate = (element) => {
+  const html = htmlFor(element);
+  return html`
+    <div class="i-amphtml-story-interactive-poll-container">
+      <div class="i-amphtml-story-interactive-prompt-container"></div>
+      <div class="i-amphtml-story-interactive-option-container"></div>
+    </div>
+  `;
+};
+
+/**
+ * Generates the template for each option.
+ *
+ * @param {!Element} option
+ * @return {!Element}
+ */
+const buildOptionTemplate = (option) => {
+  const html = htmlFor(option);
+  return html`
+    <span class="i-amphtml-story-interactive-option">
+      <span class="i-amphtml-story-interactive-option-text"></span>
+      <span class="i-amphtml-story-interactive-option-percentage">
+        <span class="i-amphtml-story-interactive-option-percentage-text"></span>
+        <span class="i-amphtml-story-interactive-option-percentage-sign"
+          >%</span
+        >
+      </span>
+    </span>
+  `;
+};
+
+export class AmpStoryInteractivePoll extends AmpStoryInteractive {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element, InteractiveType.POLL, [2, 4]);
+  }
+
+  /** @override */
+  buildCallback() {
+    super.buildCallback(CSS);
+  }
+
+  /** @override */
+  buildComponent() {
+    this.rootEl_ = buildPollTemplate(this.element);
+    this.attachContent_(this.rootEl_);
+    return this.rootEl_;
+  }
+
+  /**
+   * Finds the prompt and options content
+   * and adds it to the quiz element.
+   *
+   * @private
+   * @param {Element} root
+   */
+  attachContent_(root) {
+    this.attachPrompt_(root);
+    this.options_.forEach((option, index) =>
+      this.configureOption_(option, index)
+    );
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the quiz element.
+   *
+   * @param {!./amp-story-interactive.OptionConfigType} option
+   * @param {number} index
+   * @private
+   */
+  configureOption_(option, index) {
+    const convertedOption = buildOptionTemplate(this.element);
+    convertedOption.optionIndex_ = index;
+
+    // Extract and structure the option information
+    convertedOption.querySelector(
+      '.i-amphtml-story-interactive-option-text'
+    ).textContent = option.text;
+
+    this.rootEl_
+      .querySelector('.i-amphtml-story-interactive-option-container')
+      .appendChild(convertedOption);
+  }
+
+  /**
+   * @override
+   */
+  updateOptionPercentages_(optionsData) {
+    if (!optionsData) {
+      return;
+    }
+
+    const percentages = this.preprocessPercentages_(optionsData);
+
+    percentages.forEach((percentage, index) => {
+      const currOption = this.getOptionElements()[index];
+      currOption.querySelector(
+        '.i-amphtml-story-interactive-option-percentage-text'
+      ).textContent = `${percentage}`;
+      setStyle(currOption, '--option-percentage', percentages[index] + '%');
```

Comment:
What is the benefit of editing a variable so that a pseudo element updates?
Why not have a percentage bar dom element and set its style?

Reply:
RTL layouts use this variable in a different way (the percentage fills in from the other side), so having this level of indirection gets rid of logic in JS to account for those cases. This variable is then used on the option::after element depending on the direction. Quizzes also do it this way, because the RTL layout changes the translateX.